### PR TITLE
enable PGD masks for non-classification tasks

### PR DESCRIPTION
**Pull request by mathsinn**
Signed-off-by: Mathieu Sinn <mathsinn@ie.ibm.com>

# Description

Disabled error that was thrown when providing masks to PGD attacks for non-classification estimators.

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] Local tests

**Test Configuration**:
- MacOS
- Python version 3.7
- ART version or commit number 1.4.1
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi @mathsinn Thank you for this pull request. It looks good to me.

Should we remove the argument `classifier_mixin` from the method `_get_mask` and its docstring?

**Comment by mathsinn** [comment]
Hi @beat-buesser  you're right, we should do that - I've pushed that change.

**Review by beat-buesser — APPROVED** [review]
